core/src/main/java/org/apache/calcite/rel/rules/PushProjector.java

```diff
@@ -250,6 +251,33 @@ public PushProjector(
         strongBitmap = ImmutableBitSet.range(nSysFields, nChildFields);
       }
 
+    } else if (childRel instanceof Correlate) {
+      Correlate joinRel = (Correlate) childRel;
+      List<RelDataTypeField> leftFields =
+          joinRel.getLeft().getRowType().getFieldList();
+      List<RelDataTypeField> rightFields =
+          joinRel.getRight().getRowType().getFieldList();
+      nFields = leftFields.size();
+      nFieldsRight = rightFields.size();
+      nSysFields = 0;
+      childBitmap =
+          ImmutableBitSet.range(nSysFields, nFields + nSysFields);
+      rightBitmap =
+          ImmutableBitSet.range(nFields + nSysFields, nChildFields);
+
+      // Required columns need to be included in project
+      projRefs.or(BitSets.of(joinRel.getRequiredColumns()));
+
+      switch (joinRel.getJoinType()) {
```

Comment:
In the Correlate class [1] I also see SEMI and ANTI join types.  Should those be handled here explicitly or go through the default handling ? 


[1] https://github.com/apache/calcite/blob/master/core/src/main/java/org/apache/calcite/rel/core/Correlate.java#L148